satpy/readers/hsaf_h5.py

```diff
@@ -0,0 +1,142 @@
+        super(HSAFFileHandler, self).__init__(filename,
+                                              filename_info,
+                                              filetype_info)
+
+        self._msg_datasets = {}
+        self._h5fh = h5py.File(self.filename, 'r')
+        self._data_time = self.filename_info['sensing_time']
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        return self._data_time + timedelta(hours=23, minutes=59, seconds=59)
+
+    def _prepare_variable_for_palette(self, msg, ds_info):
+        colormap = np.array(np.array(msg))
+        return xr.DataArray(colormap, attrs=ds_info, dims=('idx', 'RGB'))
+
+    def get_metadata(self, msg, name):
+        """Get the metadata."""
+        if name == 'SC':
+            ds_info = {
+                'name': 'SC',
+                'filename': self.filename,
+                'data_time': self.start_time,
+                'nx': msg.shape[1],
+                'ny': msg.shape[0]
+            }
+        elif name == 'SC_pal':
+            ds_info = {
+                'name': 'SC_pal'
+            }
+        return ds_info
+
+    def get_area_def(self, dsid):
+        """Area definition for h10 SC dataset.
+
+        Since it is not available in the HDF5 message,
+        using hardcoded one (it's known).
+        """
+        if dsid['name'] == 'SC':
+            return self._get_area_def()
+        raise NotImplementedError
+
+    def _get_area_def(self):
+        """Area definition for h10 - hardcoded.
+
+        Area definition not available in the HDF5 message,
+        so using hardcoded one (it's known).
+
+        hsaf_h10:
+          description: H SAF H10 area definition
+          projection:
+            proj: geos
+            lon_0: 0
+            h: 35785831
+            x_0: 0
+            y_0: 0
+            a: 6378169
+            rf: 295.488065897001
+            no_defs: null
+            type: crs
+          shape:
+            height: 916
+            width: 1902
+          area_extent:
+            lower_left_xy: [-1936760.3163240477, 2635854.280233425]
+            upper_right_xy: [3770006.7195370505, 5384223.683413638]
+            units: m
+        """
+        fd_def = get_area_def('msg_seviri_fes_3km')
+        hsaf_def = fd_def[area_y_offset:area_y_offset+916,
+                          area_x_offset:area_x_offset+1902]
+
+        return hsaf_def
+
+    def _get_dataset(self, ds_name):
+        if ds_name == 'SC_pal':
+            _ds_name = 'colormap'
+        else:
+            _ds_name = ds_name
+        return self._h5fh.get(_ds_name)
+
+    def get_dataset(self, ds_id, ds_info):
+        """Read a HDF5 file into an xarray DataArray."""
+        variable = None
+        ds = self._get_dataset(ds_id['name'])
+        ds_info = self.get_metadata(ds, ds_id['name'])
+
+        if ds_id['name'] == 'SC':
+            ds_info['start_time'] = self._data_time
+            ds_info['data_time'] = self._data_time
```

Comment:
Don't remember where these are used, but should the end time also be here?
```suggestion
            ds_info['start_time'] = self.start_time
            ds_info['data_time'] = self.start_time
```